test/functional/test_framework/wallet.py

```diff
@@ -119,12 +119,14 @@ def _bulk_tx(self, tx, target_weight):
     def get_balance(self):
         return sum(u['value'] for u in self._utxos)
 
-    def rescan_utxos(self):
+    def rescan_utxos(self, *, include_mempool=True):
         """Drop all utxos and rescan the utxo set"""
         self._utxos = []
         res = self._test_node.scantxoutset(action="start", scanobjects=[self.get_descriptor()])
         assert_equal(True, res['success'])
         for utxo in res['unspents']:
+            if not include_mempool and not self._test_node.gettxout(txid=utxo["txid"], n=utxo["vout"]):
```

Comment:
This just seems to exclude utxos if they are spent in mempool transactions. Shouldn't it also include utxos if they are created in mempool transactions?

Reply:
[test_sequence_lock_confirmed_inputs](https://github.com/bitcoin/bitcoin/blob/69f0f6bc5ec9177cde7fdd0e73dac470f5641f82/test/functional/feature_bip68_sequence.py#L146-L156) would choose a random number of utxos from get_utxos, and if I included the transaction that is inside the mempool, the txn-mempool-conflict would occur.

Reply:
I mean that someone setting the bool on the method would expect that this includes utxos created in the mempool. Also, I fail to see how including utxos created in the mempool can cause a conflict, as long as you remove utxos spent in the mempool.

This was also the behavior previously without MiniWallet, so it seems better to keep it, no?

Might be possible by calling scan_tx on all mempool txs, assuming they are topologically sorted.

Reply:
Or should I use `getrawtransaction` to scan mempool, but this would require `txindex=1`.

Reply:
`getrawmempool` does not require txindex

Reply:
`getrawmempool` returns only the `transactionid` but not the `tx.vin`, which is required to scan spent UTXOs.

Reply:
`getrawmempool` can return the transaction hex-encoded. You can decode it, or you may be able to call `getmempoolentry` if you need it decoded.

Reply:
I'll see what I can do tomorrow morning, thanks!

Reply:
> `getrawmempool` can return the transaction hex-encoded. You can decode it, or you may be able to call `getmempoolentry` if you need it decoded.

Sorry to bother you, but I just checked the bitcoin reference with [getrawmempool](https://developer.bitcoin.org/reference/rpc/getrawmempool.html) and [getmempoolentry](https://developer.bitcoin.org/reference/rpc/getmempoolentry.html), and there is no way to get `tx.vin` with these two methods.

Reply:
Right, but `getrawtransaction` can